pandas/core/arrays/datetimes.py

```diff
@@ -99,31 +100,40 @@ def wrapper(self, other):
         meth = getattr(dtl.DatetimeLikeArrayMixin, opname)
 
         if isinstance(other, (datetime, np.datetime64, compat.string_types)):
-            if isinstance(other, datetime):
+            if isinstance(other, (datetime, np.datetime64)):
```

Comment:
since we do this all over the place, thing about making a type for this (for in internal use only), and/or maybe a ``is_datetime_scalar`` or something, just so we are consistent

Reply:
I've got a branch going looking at `core.dtypes.common` (inspired by noticing the footgun status if `is_datetimelike`), will look at this there so I can get all the places it is used in one swoop.